sql/size_utils.sql

```diff
@@ -208,6 +210,9 @@ BEGIN
                 END IF;
         END IF;
 
+        /* Lock hypertable to avoid risk of concurrent process dropping chunks */
+        EXECUTE format('LOCK TABLE %I.%I IN ACCESS SHARE MODE', schema_name, table_name);
```

Comment:
Yes this is correct, Postgres does not allow an ACCESS SHARE lock on the table without SELECT privilege.

If we don't take a lock here then it could be a race condition with parallel connection dropping the chunk while the `hypertable_size` calculation is in progress.